src/test/java/uk/gov/hmcts/reform/prl/repositories/CcdCaseRepositoryTest.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.prl.repositories;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.EventRequestData;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.services.citizen.CaseService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.prl.enums.CaseEvent.LINK_CITIZEN;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CcdCaseRepositoryTest {
+
+    @InjectMocks
+    CcdCaseRepository ccdCaseRepository;
+
+    @InjectMocks
+    CaseService caseService;
+
+    @Mock
+    CcdCaseApi ccdCaseApi;
+
+    @Mock
+    CaseRepository caseRepository;
+
+    public static final String authToken = "Bearer TestAuthToken";
+
+    public static final String s2sToken = "s2s AuthToken";
+
+    private String caseId = "";
+
+    private CaseData caseData;
+
+    private CaseDetails emptyCaseDetails;
+
+    @Before
+    public void setup() {
+        caseId = "test_case";
+        caseData = CaseData.builder().build();
+        emptyCaseDetails = CaseDetails.builder().build();
+    }
+
+    @Test
+    public void testLinkDefendant() throws JsonProcessingException {
```

Comment:
can we add one negative scenario as well?

Reply:
negetive scenario added.